### ISSUE TITLE
QOpenGLShader::compile - segfault

### ISSUE THREAD

**Issue by FidgetteSpinneur** [spyder-ide/spyder#8002]
## Problem Description

Attempting to run Spyder results in segfault, see below for details.

### Paste Traceback/Error Below (if applicable)
```
$ SPYDER_DEBUG=3 spyder
Start of MainWindow constructor
End of MainWindow constructor
*** Start of MainWindow setup ***
  ..core actions
  ..toolbars
  ..tools
  ..sift?
  ..plugin: internal console
    ..internal console: initializing
  ..plugin: working directory
Loading help...
Loading outline explorer...
Loading editor...
Introspection Plugin Loaded: rope
Introspection Plugin Loaded: jedi
Introspection Plugin Loaded: fallback
setfocusto: <spyder.widgets.sourcecode.codeeditor.CodeEditor object at 0x7f0d24631048>
current_changed: 0 <spyder.widgets.sourcecode.codeeditor.CodeEditor object at 0x7f0d24631048> 139694421774408
Added thread <spyder.widgets.editor.AnalysisThread object at 0x7f0d2463d558> to queue
Added thread <spyder.widgets.editor.AnalysisThread object at 0x7f0d2463d678> to queue
  ..widgets
Loading file explorer...
Loading history plugin...
Updating queue:
    started: 0
    pending: 2
===>starting: <spyder.widgets.editor.AnalysisThread object at 0x7f0d2463d558>
Updating queue:
    started: 1
    pending: 1
===>starting: <spyder.widgets.editor.AnalysisThread object at 0x7f0d2463d678>
Loading online help...
Updating queue:
    started: 0
    pending: 0
Updating queue:
    started: 0
    pending: 0
Loading project explorer...
Loading namespace browser...
Loading IPython console...
Setting up main window...
Setting up window...
*** End of MainWindow setup ***
Call to 'close_threads'
Initialized fallback
setfocusto: <spyder.widgets.sourcecode.codeeditor.CodeEditor object at 0x7f0d24638318>
current_changed: 0 <spyder.widgets.sourcecode.codeeditor.CodeEditor object at 0x7f0d24638318> 139694421803800
QOpenGLShader::compile(Vertex): failed
QOpenGLShader::compile(Fragment): failed
Could not link shader program:
 ""
Segmentation fault (core dumped)



```

## Versions

* Spyder version: 3.3.1
* Python version: 3.6.6 
* Qt version: 5.9.6
* PyQt version: 5.9.2
* Operating System name/version: Ubuntu 16.04
* Linux kernel 4.4.0-137-generic
* NVidia drivers: 410.48

Any hints?
Thanks,
Ben

**Comment by FidgetteSpinneur** [comment]
> > NVidia drivers: 410.48
> 
> Could you use your Intel card by default and try again? Thanks!

Actually, we're shouldn't even be using any GPU drivers. It is a shared box which is remotely accessed (VNC mostly) and used for computation (hence the Nvidia card).

The above dump is from VNC -- when launching Spyder over SSH forwarding, it still segfaults, but the dump looks somewhat different:

```
$ SPYDER_DEBUG=3 spyder
Start of MainWindow constructor
End of MainWindow constructor
*** Start of MainWindow setup ***
  ..core actions
  ..toolbars
  ..tools
  ..sift?
  ..plugin: internal console
    ..internal console: initializing
  ..plugin: working directory
Loading help...
WebEngineContext used before QtWebEngine::initialize() or OpenGL context creation failed.
Loading outline explorer...
Loading editor...
Introspection Plugin Loaded: rope
Introspection Plugin Loaded: jedi
Introspection Plugin Loaded: fallback
setfocusto: <spyder.widgets.sourcecode.codeeditor.CodeEditor object at 0x7f1578403048>
current_changed: 0 <spyder.widgets.sourcecode.codeeditor.CodeEditor object at 0x7f1578403048> 139730188513352
Added thread <spyder.widgets.editor.AnalysisThread object at 0x7f157840f558> to queue
Added thread <spyder.widgets.editor.AnalysisThread object at 0x7f157840f678> to queue
  ..widgets
Loading file explorer...
Loading history plugin...
Loading online help...
Loading project explorer...
Updating queue:
    started: 0
    pending: 2
===>starting: <spyder.widgets.editor.AnalysisThread object at 0x7f157840f558>
Updating queue:
    started: 1
    pending: 1
===>starting: <spyder.widgets.editor.AnalysisThread object at 0x7f157840f678>
Loading namespace browser...
Updating queue:
    started: 0
    pending: 0
Updating queue:
    started: 0
    pending: 0
Loading IPython console...
Setting up main window...
Initialized fallback
Setting up window...
*** End of MainWindow setup ***
Failed to create OpenGL context for format QSurfaceFormat(version 2.0, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize 24, redBufferSize -1, greenBufferSize -1, blueBufferSize -1, alphaBufferSize -1, stencilBufferSize 8, samples 0, swapBehavior QSurfaceFormat::SwapBehavior(DefaultSwapBehavior), swapInterval 1, profile  QSurfaceFormat::OpenGLContextProfile(NoProfile))
Aborted (core dumped)
```

As I understand, the Nvidia drivers shouldn't be involved here, correct?

Thanks,
Ben

**Comment by ccordoba12** [comment]
> The above dump is from VNC -- when launching Spyder over SSH forwarding, it still segfaults, but the dump looks somewhat different:

This error will be fixed in our next version (3.3.2). See issue #7447 for the details.
